keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,738 @@
+    | name: bar-$uuid                      |
+    | spec:                                |
+    |   bucketAccessClassName: bac1        |
+    |   parameters:                        |
+    |     key: value                       |
+    |   bucketName: br1-$uuid              | 
+    | status:                              |
+    |   credentialsSecret: nil             |
+    |   conditions:                        |
+    |   - type: AccessGranted              |
+    |     status: False                    |
+    |--------------------------------------|
+```
 
-If the bucket sharing is all within the same namespace then each `BucketAccessRequest` (in the namespace) only needs to reference the existing BR.
+###### 3. COSI calls driver to create BucketAccess
 
-Here is the workflow:
+The COSI sidecar, which runs alongside each of the drivers, listens for BucketAccess events. All but the sidecar with the specified provisioner will ignore the BucketAccess object. Only the appropriate sidecar will make a gRPC request to its driver to generate credentials for a Bucket in the OSP.
 
-+ A `Bucket` instance created by COSI must be cloned by the admin (each `Bucket` instance needs a unique name).
-+ The user creates a `BucketRequest` (BR) that specifies the `Bucket` instance name (BR.bucketInstanceName) provided by the admin.
-+ The user creates a `BucketAccessRequest` (BAR) - see [grant bucket access](#grant-bucket-access) -- and the BAR references their BR.
+More information about COSI gRPC API is [here](#cosi-grpc-api)
 
-## Delete Bucket
+```
+    COSI Driver (s3.amazonaws.com)
+    |------------------------------------------|
+    | grpc ProvisionerGrantBucketAccess({      |
+    |      "name": "bar-$uuid",                |
+    |      "bucketID": "br-$uuid",             |
+    |      "parameters": {                     |
+    |         "key": "value"                   |
+    |      }                                   |
+    | })                                       |
+    |------------------------------------------|
+```
+Each BucketAccess is meant to map to one account (or its equivalent) in the OSP with privileges to access the bucket. Once the credentials are successfully generated in the OSP, a secret by the name specified in the BucketAccessRequest is created and the credential values are stored in it. The secret resides in the namespace of the BucketAccessRequest. 
 
-![DeleteBucket Workflow](images/delete-bucket.png)
+The sidecar sets the `AccessGranted` condition to `True` in the BucketAccess object and sets the `credentialsSecret` field to the name and namespace of the generated secret. At this point, the BucketAccess is ready to be utilized by workloads.
 
-This workflow describes the automation designed for deleting a `Bucket` instance and optionally the related backend bucket. Revoking access to this bucket is covered in [Revoke Bucket Access](#revoke-bucket-access).
+NOTE: The secret is not created until the credentials have been generated.
```

Comment:
Since the sidecar is not expected to have access to user namespaces, I assume the secret created by the sidecar lives in the provisioner's ns. So the secret is not created in the BAR's ns until the workload pod starts up (via the cois-node-adapter). Is this still correct?

Reply:
This may reasonably be an implementation detail, but how is idempotency handled with if the secret name already exists in either the user's or provisioner's ns?